File: src/DataLoader.m
function [Data, Labels] = DataLoader(foldername)
    fname = '%s_SURF_L10.mat';
    domain_names = {'amazon', 'webcam', 'dslr', 'Caltech10'};

    Data = cell(numel(domain_names));
    Labels = cell(numel(domain_names));
    for d = 1:numel(domain_names)
       fullfilename = fullfile(foldername, sprintf(fname, domain_names{d}));
       load(fullfilename);
       fts = NormData(fts);
       Data{d} = fts;
       Labels{d} = labels';
    end
end

function fts = NormData(fts)
    fts = fts ./ repmat(sqrt(sum(fts.^2,2)),1,size(fts,2));
    % mean = 0 and variance = 1
    fts = zscore(fts,1); 
end

File: College_Project/Final_eval/src/dataloader.py
import scipy.io
from misc import NormData

def LoadOfficePlusCaltechData(foldername):
    domain_names = ['amazon_SURF_L10.mat', 'webcam_SURF_L10.mat', 'dslr_SURF_L10.mat', 'Caltech10_SURF_L10.mat']
    Data = []
    Labels = []
    for idx, name in enumerate(domain_names):
        fullfilename = foldername + name
        obj = scipy.io.loadmat(fullfilename)
        
        fts = obj['fts']
        labels = obj['labels']
        fts = NormData(fts)
        Data.append(fts)
        Labels.append(labels)
        
    return Data, Labels